rag/advanced_rag/knowlege_compile/lightgraph.py

```
@@ -0,0 +1,744 @@
+    ``run_document_post_chunking_if_last()`` — the last-chunk-task
+    gating is already done by the caller.
+
+    Args:
+        handler: TaskHandler instance (provides _load_chunks_for_doc)
+        ctx: TaskContext with doc_id, kb_id, tenant_id, language, etc.
+        embedding_model: LLMBundle for generating embeddings.
+    """
+    tenant_id, kb_id, doc_id = ctx.tenant_id, ctx.kb_id, ctx.doc_id
+    language = ctx.language or "en"
+    progress_cb = ctx.progress_cb
+    cancel_check = lambda: ctx.has_canceled_func(ctx.id)
```

Ruff (0.15.20)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `cancel_check` as a `def`

(E731)